app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/partnership/PartnershipOtherPartnerContactNameController.scala

```diff
@@ -54,80 +54,80 @@ class PartnershipOtherPartnerContactNameController @Inject() (
 
   def displayPage(): Action[AnyContent] =
     (authenticate andThen journeyAction) { implicit request =>
+      // TODO will have been populated by the preceding GRS journey
       val inflightOtherPartner =
         Some(
           Partner(organisationName = Some("Partner organisation"),
                   partnerType = Some(PartnerTypeEnum.SOLE_TRADER)
-          ) // TODO will have been populated by the preceding GRS journey
+          )
         )
 
-      (for {
-        inflight         <- inflightOtherPartner
-        organisationName <- inflight.organisationName
-      } yield (for {
-        contactDetails <- inflight.contactDetails
-        firstName      <- contactDetails.firstName
-        lastName       <- contactDetails.lastName
+      inflightOtherPartner.map { inflight =>
+        inflight.organisationName.map { organisationName =>
+          val existingNameFields = for {
+            contactDetails <- inflight.contactDetails
+            firstName      <- contactDetails.firstName
+            lastName       <- contactDetails.lastName
+          } yield (firstName, lastName)
+
+          val form = existingNameFields match {
+            case Some(data) =>
+              MemberName.form().fill(MemberName(data._1, data._2))
+            case None =>
+              MemberName.form()
+          }
 
-      } yield (firstName, lastName)) match {
-        case Some(data) =>
-          Ok(
-            page(MemberName.form().fill(MemberName(data._1, data._2)),
-                 organisationName,
-                 partnershipRoutes.PartnershipPartnersListController.displayPage(),
-                 partnershipRoutes.PartnershipOtherPartnerContactNameController.submit()
-            )
-          )
-        case None =>
           Ok(
-            page(MemberName.form(),
+            page(form,
                  organisationName,
                  partnershipRoutes.PartnershipPartnersListController.displayPage(),
                  partnershipRoutes.PartnershipOtherPartnerContactNameController.submit()
             )
           )
-      }).getOrElse(NotFound)
+
+        }.getOrElse(NotFound)
+      }.getOrElse(NotFound)
```

Comment:
I think elsewhere we just display the page with an empty form if the data used to populate the form is not present in the registration. Not a showstopper but it feels a little "different" to the other controllers.

Reply:
Yeah... one of the pages has a title dependant on the input from the previous page.
ie. Enter John Smith's phone number. 
That's the bit which tipped in over the edge.